addons/project/tests/test_project_sharing_portal_access.py

```diff
@@ -76,3 +76,30 @@ def test_write_with_portal_user(self):
         for field in self.other_fields_task:
             with self.assertRaises(AccessError):
                 self.task_portal.with_user(self.user_portal).write({field: 'dummy'})
+
+    def test_wizard_confirm(self):
+        partner_portal_no_user = self.env['res.partner'].create({
+            'name': 'NoUser portal',
+            'email': 'no@user.portal',
+            'company_id': False,
+            'user_ids': []})
+
+        project_share_wizard_no_user = self.env['project.share.wizard'].create({
+            'access_mode': 'edit',
+            'res_model': 'project.project',
+            'res_id': self.project_portal.id,
+            'partner_ids': [
+                Command.link(partner_portal_no_user.id),
+            ],
+        })
+        self.env["res.config.settings"].create({
+            "auth_signup_uninvited": 'b2b',
+        }).execute()
+        project_share_wizard_no_user_action = project_share_wizard_no_user.action_share_record()
+        self.assertEqual(project_share_wizard_no_user_action['type'], 'ir.actions.act_window', 'Sharing a project with partner without user should display a confimation dialog')
+        project_share_wizard_confirmation = self.env['project.share.wizard'].browse(project_share_wizard_no_user_action['res_id'])
```

Comment:
I am wondering whether ``project_share_wizard_no_user_action`` and ``project_share_wizard_confirmation`` are the same thing because they are both the confirmation dialog :thinking:

Reply:
So basically
`project_share_wizard_no_user_action` =>
![image](https://github.com/odoo/odoo/assets/36259462/2e188cd8-8289-4452-ae6d-d1891f4d634c)

It's the wizard generated in which you can chose the partner to who you want to share the project/let create a portal user

`project_share_wizard_confirmation` =>
![image](https://github.com/odoo/odoo/assets/36259462/726ff749-8bb0-4e11-9007-944b349787dc)

It's the wizard that warns you you're about to grant portal right to a partner which didn't have any before (doesn't appear if the partner already had access to another project for exemple)

So you have to confirm both for the link to be sent in our case